feat: Adds the MetadataBar to the Explore header

michael-s-molina opened the pull request:
### SUMMARY
- Adds the `MetadataBar` to the Explore header
- Improves the `MetadataBar` resize capabilities and changes the component to work with server-side timestamps
- Changes the logic to get slice-related dashboards to use them in the metadata bar
- Adds and adjusts tests

@kasiazjc I didn't add a tooltip for the number of dashboards because it requires a future PR that will increment the 3 dots menu with the list of related dashboards. This will be made in a follow-up.

@codyml @rusackas 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/191856911-45de56e6-a020-4178-b600-705bb1bf6538.mov

### TESTING INSTRUCTIONS
- Check the video for instructions on the general behavior
- Make sure that the previous logic of inheriting dashboard color schemes is not affected

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

michael-s-molina committed:
feat: Adds the MetadataBar to the Explore header

michael-s-molina commented on superset-frontend/src/explore/components/ExploreChartHeader/index.jsx:
```diff
@@ -75,51 +77,51 @@ export const ExploreChartHeader = ({
-        }
-      })
-      .catch(() => {});
+  const updateCategoricalNamespace = async () => {
```
@geido Can you please check if I didn't break anything here?

michael-s-molina commented:
/testenv up

geido commented:
Hey @michael-s-molina a few quick comments from a first look

Should we add a placeholder text when there are no available owners?

![Screenshot 2022-09-26 at 17 46 53](https://user-images.githubusercontent.com/60598000/192308010-376b7f55-bdb6-44ea-bef1-a343fab7fe26.png)

Spacing requires some improvement

![Screenshot 2022-09-26 at 17 51 09](https://user-images.githubusercontent.com/60598000/192309169-dcc4a961-d894-406f-8070-4a38f19a919b.png)

Can we make the empty space equal on both ends?

![Screenshot 2022-09-26 at 17 53 53](https://user-images.githubusercontent.com/60598000/192309872-b064ebe9-2626-4d5b-9d2b-73a773ba6f8f.png)

michael-s-molina committed:
Addresses comments

michael-s-molina commented:
> Should we add a placeholder text when there are no available owners?

Added `None` for this case. I thought `Not available` does not apply here because we have the information.

> Spacing requires some improvement

Fixed

> Can we make the empty space equal on both ends?

This is a little more complicated. Currently, we have a minimum width of `70px` defined for metadata items according to the design system specs. We use this minimum width to calculate the breakpoint that will trigger the icons-only mode. We would need to define another way of calculating the breakpoint and update the specs. I'm pinging @kasiazjc for awareness here. I'll open a ticket for this and work on it in a separate PR because of the required changes.

geido commented:
/testenv up

geido reviewed (approved):
Tested locally. LGTM!
